test/libsolidity/semanticTests/inlineAssembly/calldata_struct_assign_and_return.sol

```diff
@@ -0,0 +1,17 @@
+contract C{
+  struct S{ int8 x; int8 y; }
+  function f() internal pure returns(S calldata s) {
+    assembly {
+      s := 0x24
+    }
+  }
```

Comment:
This could actually be quite a useful pattern and replace using abi decoding from ``msg.data``...
We do allow this for storage references, so I see no reason for not allowing it for calldata.

Reply:
Although... hm... I'm not entirely sure, this might bypass validation on the outermost layer, let me see...

Reply:
Well, we do allow it for dynamic calldata arrays anyways, so...